packages/react/src/components/form/form.stories.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  Asterisk,
+  Button,
+  Checkbox,
+  Field,
+  FieldLabel,
+  Form,
+  FormProps,
+  HelperText,
+  Input,
+  Radio,
+  Validation,
+} from '@onfido/castor-react';
+import React from 'react';
+import { Meta, Story } from '../../../../../docs';
+
+export default {
+  title: 'React/Form',
+  component: Form,
+} as Meta<FormProps<unknown>>;
+
+interface Values {
+  firstName: string;
+  lastName: string;
+  email: string;
+  day?: number;
+  month?: number;
+  year?: number;
+  source?: string;
+  agreement: boolean;
+}
+
+export const Playground: Story<FormProps<Values>> = (props) => (
+  <Form {...props}>
+    <Field>
+      <Input name="firstName" required>
+        First name <Asterisk aria-label="required" />
+      </Input>
+      <Validation state="error" for="valid">
+        Please fill in this field
+      </Validation>
+    </Field>
+
+    <Field>
+      <Input name="lastName" required>
+        Last name <Asterisk aria-label="required" />
+      </Input>
+      <Validation state="error" for="valid">
+        Please fill in this field
+      </Validation>
+    </Field>
+
+    <Field>
+      <Input type="email" name="email" required>
+        Email address <Asterisk aria-label="required" />
+      </Input>
+      <Validation state="error" for="valueMissing">
+        Please fill this field
+      </Validation>
+      <Validation state="error" for="typeMismatch">
+        Please enter a valid email address
+      </Validation>
+    </Field>
+
+    <Field>
+      <FieldLabel>
+        Date of birth
+        <HelperText>DD / MM / YYYY</HelperText>
+        <div
+          style={{
+            display: 'grid',
+            gap: '2rem',
+            grid: '1fr / 5rem 5rem 6rem',
+          }}
+        >
+          <Input type="number" name="day" placeholder="DD" min={1} max={31} />
+          <Input type="number" name="month" placeholder="MM" min={1} max={12} />
+          <Input
+            type="number"
+            name="year"
+            placeholder="YYYY"
+            min={new Date().getUTCFullYear() - 150}
+            max={new Date().getUTCFullYear()}
+          />
+        </div>
+      </FieldLabel>
```

Comment:
FieldLabel is not designed to have Input or Textarea nested also within a `<div>` (or anything else).

You end up not having a required space between label text and input(s).

For such you instead should do like this:

```suggestion
      <FieldLabel htmlFor="dob-day">
        Date of birth
        <HelperText>DD / MM / YYYY</HelperText>
      </FieldLabel>
      <div
          style={{
            display: 'grid',
            gap: '2rem',
            grid: '1fr / 5rem 5rem 6rem',
          }}
        >
          <Input type="number" id="dob-day" name="day" placeholder="DD" min={1} max={31} />
          <Input type="number" name="month" placeholder="MM" min={1} max={12} />
          <Input
            type="number"
            name="year"
            placeholder="YYYY"
            min={new Date().getUTCFullYear() - 150}
            max={new Date().getUTCFullYear()}
          />
        </div>
```

Reply:
This is a tricky one altogether, because the both `Field` and the label can't work for all inputs at the same time.

We have to think this pattern through, but for now your suggestion makes sense, presumably you start by typing the day.